ios/Classes/VICallModule.m

```diff
@@ -551,17 +568,100 @@ - (void)didDetectVoiceActivityStop:(VIEndpoint *)endpoint {
+        @"codec": codec ? codec : [NSNull null],
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
+- (void)call:(VICall *)call didDetectLocalVideoDegradation:(CGSize)actualSize targetSize:(CGSize)targetSize issueLevel:(VIQualityIssueLevel)level {
+    NSDictionary *actualSizeStruct = @{
+        @"width": @(actualSize.width),
+        @"height": @(actualSize.height)
+    };
+    NSDictionary *targetSizeStruct = @{
+        @"width": @(targetSize.width),
+        @"height": @(targetSize.height)
+    };
+    [self sendQualityIssueEvent:@{
+        @"event": @"VIQualityIssueTypeLocalVideoDegradation",
+        @"actualSizeStruct": actualSizeStruct,
+        @"targetSizeStruct": targetSizeStruct,
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
+- (void)call:(VICall *)call didDetectIceDisconnected:(VIQualityIssueLevel)level {
+    [self sendQualityIssueEvent:@{
+        @"event": @"VIQualityIssueTypeIceDisconnected",
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
+- (void)call:(VICall *)call didDetectHighMediaLatency:(NSTimeInterval)latency issueLevel:(VIQualityIssueLevel)level {
+    [self sendQualityIssueEvent:@{
+        @"event": @"VIQualityIssueTypeHighMediaLatency",
+        @"latency": @([[NSNumber fromTimeInterval:latency] doubleValue]),
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
+- (void)call:(VICall *)call didDetectNoAudioSignal:(VIQualityIssueLevel)level {
+    [self sendQualityIssueEvent:@{
+        @"event": @"VIQualityIssueTypeNoAudioSignal",
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
+- (void)                   call:(VICall *)call
+didDetectNoAudioReceiveOnStream:(VIRemoteAudioStream *)audioStream
+                   fromEndpoint:(VIEndpoint *)endpoint
+                     issueLevel:(VIQualityIssueLevel)level {
+    [self sendQualityIssueEvent:@{
+        @"event": @"VIQualityIssueTypeNoAudioReceive",
+        @"audiostreamId": audioStream.streamId,
+        @"endpointId": endpoint.endpointId,
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
+- (void)                   call:(VICall *)call
+didDetectNoVideoReceiveOnStream:(VIRemoteVideoStream *)videoStream
+                   fromEndpoint:(VIEndpoint *)endpoint
+                     issueLevel:(VIQualityIssueLevel)level {
+    [self sendQualityIssueEvent:@{
+        @"event": @"VIQualityIssueTypeNoVideoReceive",
+        @"videostreamId": videoStream.streamId,
+        @"endpointId": endpoint.endpointId,
+        @"issueLevel": [VoximplantUtils convertQualityIssueLevelToInt:level]
+    }];
+}
+
 #pragma mark - FlutterStreamHandler
 
 - (FlutterError * _Nullable)onCancelWithArguments:(id _Nullable)arguments {
     self.eventSink = nil;
+    self.eventSinkQualityIssues = nil;
```

Comment:
please check which one is canceled before setting it to nil

Reply:
done